### DISCUSSION TITLE
Video Processing Library in dotnet core

### DISCUSSION THREAD

**Discussion by Junaid-Amjad** [AvaloniaUI/Avalonia#5445] [Q&A] [answered]
I know it is not a right question in this forum but it would be helpful if i get to know about video processing library used/worked in Avalonia or dotnet core? It is just a general question.

**Comment by kekekeks** [comment, marked as answer]
ffmpeg should work fairly well. Consider using https://github.com/Ruslan-B/FFmpeg.AutoGen for interop.

**Reply by Junaid-Amjad** [reply]
Okay, i think i didn't elaborate my video processing. I want to display the Live Stream in wpf application and later save the stream in video(where ffmpeg will be useful). In wpf it can be achieved using mediaelement but it is not exist in the Avalonia-WPF. Is there any alternative library which can be used in Avalonia

**Reply by kekekeks** [reply]
You can use ffmpeg and output the stream into WritableBitmap or draw via ICustomDrawingOperation for better performance.

**Reply by Junaid-Amjad** [reply]
performance is the first pirority. Is there any link where i get to know about ICustomDrawingOperation.

**Reply by kekekeks** [reply]
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/RenderDemo/Pages/CustomSkiaPage.cs

Note that `Render(IDrawingContextImpl context)` usually happens on the render thread (but sometimes can happen on the UI one), so you need to provide thread-safety locks yourself.

**Comment by grokys** [comment]
Not tried it, but there's now Avalonia suuport in LibVLCSharp I think: https://code.videolan.org/videolan/LibVLCSharp/-/issues/7

**Reply by Junaid-Amjad** [reply]
It is a good control but it don't support frame event.
